script.js

```diff
@@ -1,28 +1,97 @@
-var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
-var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
+var vals = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
+var suits = ["Clubs", "Diamonds", "Hearts", "Spades"];
 var game = {
-  deck: [],
-  players: [],
-  hands: [],
-  buildDeck: function(){
-
-  },
-  shuffleDeck: function(){
-
-  },
-  getMorePlayers: function(){
-
-  },
-  deal: function(){
-
-  },
-  findHighestCard: function(){
-
-  },
-  announceWinners: function(){
-
-  },
-  playANewGame: function(){
-
-  }
+    deck: [],
+    players: [],
+    hands: [],
+    buildDeck: function(){
+        this.deck = [];
+        for (var _ = 0; _ < suits.length; _++) {
+            this.deck.push(vals);
+        }
+    },
+    shuffleDeck: function(){
+        function shuffleArray(array) {
+            for (var i = array.length-1; i >= 0; i--) {
+                var j = Math.floor(Math.random() * i);
+                var t = array[i];
+                array[i] = array[j];
+                array[j] = t;
+            }
+            return array;
+        }
+        this.deck.forEach(function(value, index) {
+            // This action also copies to the previous element
+            // at this.deck[index-1]. Currently baffled.
+            this.deck[index] = shuffleArray(this.deck[index]);
```

Comment:
Yeah, I built the deck as an array of four arrays like: `[[1, ..., "A"], [1, ..., "A"], [1, ..., "A"], [1, ..., "A"]]` where each one's suit was determined its corresponding index in the `suit` array.

When I debug before the return of the `shuffleDeck` function, it shows a shuffled array of 12 cards like I want, but I think there's something about the assignment in line 26 that's copying the returned array to all previous elements.